src/services/completions.ts

```diff
@@ -1931,6 +1931,9 @@ namespace ts.Completions {
                 context => {
                     exportInfo.forEach(sourceFile.path, (info, symbolName, isFromAmbientModule) => {
                         if (!detailsEntryId && isStringANonContextualKeyword(symbolName)) return;
+                        // `targetFlags` should be the same for each `info`
+                        if (!isTypeOnlyLocation && !importCompletionNode && !(info[0].targetFlags & SymbolFlags.Value)) return;
+                        if (isTypeOnlyLocation && !(info[0].targetFlags & (SymbolFlags.Module | SymbolFlags.Type))) return;

```

Comment:
This is a slightly simpler, cheaper, more permissive copy of some filtering that runs on _all_ completions near the end of the pipeline in `shouldIncludeSymbol`. We theoretically should see no difference in behavior from this change because it filters out strictly less than what the later filter will remove. Replicating it sooner in the pipeline just means that we avoid doing module specifier resolution work for symbols that are going to get filtered out later. It does let through value-only modules/namespaces in type positions, which will be filtered out later, because I thought it wasn’t worthwhile to duplicate the check for whether a module/namespace contains any exported types.